haystack/nodes/connector/crawler.py

```diff
@@ -236,16 +250,31 @@ def _write_to_files(
             else:
                 text = el.text
 
-            link_split_values = link.replace("https://", "").split("/")
-            file_name = f"{'_'.join(link_split_values)}.json"
-            file_path = output_dir / file_name
-
             data = {}
             data["meta"] = {"url": link}
             if base_url:
                 data["meta"]["base_url"] = base_url
             data["content"] = text
             document = Document.from_dict(data, id_hash_keys=id_hash_keys)
+
+            param_naming = f"{link}{text}"
+            if crawler_naming_function is not None:
+                file_name_preffix_tmp = crawler_naming_function(link, text)
+                link_split_values = (
+                    file_name_preffix_tmp.replace("https://", "")
+                    .replace("http://", "")
+                    .replace("file:/", "")
+                    .replace("file://", "")
+                    .replace("\0", "")
+                    .split("/")
+                )
+                file_name_preffix = f"{'_'.join(link_split_values)}"
```

Comment:
Slash is not supported on any OS, so if we do not remove, then it'll be impossible to write almost all web or local files. On the old code only "https://" was being replaced, I changed to http and the local file pattern also. The only replacement which is a bit harder to happen is the “\0”. If we remove this, crawling will fail in almost all saves.

Reply:
For example, if we keep the current file name pattern (link) we will get errors on Windows when saving a simple web page. I think we should or clean the file name (even for AWS S3, since it doesn't allow spaces also) or change the default behavior. If the default behavior become the ID we could take the responsibility to the user naming function totally. But if we keep the link pattern, I think we should fix this as it causes issues on long URLs. The limit for the most common browsers are around 2000 chars, way higher than the 255 limit on most common file systems.

Reply:
I think this is covered by my answer above. However I'm going to ask some other colleagues of mine to check on this PR, so we get a fresh point of view. 

And if we're three we can reach an agreement by majority! :smile: